crates/contract/src/call.rs

```diff
@@ -754,4 +769,271 @@ mod tests {
             "max_priority_fee_per_gas of the transaction should be set to the right value"
         )
     }
+
+    // MULTICALL3 Tests
+    #[tokio::test]
+    async fn test_multicall() {
+        sol! {
+            #[derive(Debug, PartialEq)]
+            #[sol(rpc)]
+            interface ERC20 {
+                function totalSupply() external view returns (uint256 totalSupply);
+                function balanceOf(address owner) external view returns (uint256 balance);
+                function transfer(address to, uint256 value) external returns (bool);
+            }
+        }
+
+        const FORK_URL: &str =
+            "https://eth-mainnet.alchemyapi.io/v2/jGiK5vwDfC3F4r0bqukm-W2GqgdrxdSr";
+        let provider = ProviderBuilder::new().on_anvil_with_config(|a| a.fork(FORK_URL));
+        let weth = address!("C02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2");
+
+        let erc20 = ERC20::new(weth, &provider);
+
+        let multicall = provider
+            .multicall()
+            .add(erc20.totalSupply())
+            .add(erc20.balanceOf(address!("d8dA6BF26964aF9D7eEd9e03E53415D37aA96045")));
+
+        let (_block_num, (_ts, _bal)) = multicall.aggregate().await.unwrap();
+    }
+
+    sol! {
+        #[derive(Debug, PartialEq)]
+        #[sol(rpc)]
+        interface ERC20 {
+            function totalSupply() external view returns (uint256 totalSupply);
+            function balanceOf(address owner) external view returns (uint256 balance);
+            function transfer(address to, uint256 value) external returns (bool);
+        }
+    }
+
+    sol! {
+        // solc 0.8.25; solc DummyThatFails.sol --optimize --bin
+        #[sol(rpc, bytecode = "6080604052348015600e575f80fd5b5060a780601a5f395ff3fe6080604052348015600e575f80fd5b50600436106030575f3560e01c80630b93381b146034578063a9cc4718146036575b5f80fd5b005b603460405162461bcd60e51b815260040160689060208082526004908201526319985a5b60e21b604082015260600190565b60405180910390fdfea2646970667358221220c90ee107375422bb3516f4f13cdd754387c374edb5d9815fb6aa5ca111a77cb264736f6c63430008190033")]
+        #[derive(Debug)]
+        contract DummyThatFails {
+            function fail() external {
+                revert("fail");
+            }
```

Comment:
fyi: multicall3 does not support bubbling up of errors (or custom error forwarding for that matter) which will make it difficult to debug